### DISCUSSION TITLE
Look ahead / peek?

### DISCUSSION THREAD

**Discussion by jayhale** [pest-parser/pest#810] [Q&A] [answered]
Hello! I'm working on an SQL parser and would like to improve the approach I've taken to qualifying certain tokens. Specifically, I'd like to have some rules that look-ahead in order to match the current rule (a forward predicate).

Is there an approach to doing this in Pest?

E.g., something like below, which does not work:

```
select_key_word = { ^"SELECT" ~ &delimiter }
```

That is to say "SELECT" is a keyword when it is followed by a delimiter, otherwise it might be something else (an identifier in SQL). I want to check for the delimiter while leaving it to be matched by the next rule (most likely `WHITESPACE`).

Is there a way to do this?

**Comment by Tartasprint** [comment, marked as answer]
What is not working ? I tried [this](https://pest.rs/?g=eyJncmFtbWFyIjoiciA9IHsgKChzZWxlY3Rfa2V5X3dvcmQgfiBkZWxpbWl0ZXIgfCBzZWxlY3RfaWRlbnQpflwiO1wiKSogfiBFT0l9XG5zZWxlY3Rfa2V5X3dvcmQgPSBAeyBeXCJTRUxFQ1RcIiB-ICZkZWxpbWl0ZXIgfVxuc2VsZWN0X2lkZW50ID0gQHsgXlwiU0VMRUNUXCIgfiAhZGVsaW1pdGVyIH1cbmRlbGltaXRlciA9IHtcIixcIn1cbldISVRFU1BBQ0UgPSB7IFwiIFwiIH0iLCJpbnB1dCI6InNlbGVjdDtzZWxlY3Q7c2VsZWN0LDsifQ%3D%3D#editor) on the online editor.

It should work.
